test/cli/dag_test.go

```diff
@@ -1,28 +1,39 @@
 package cli
 
 import (
+	"os"
 	"testing"
 
 	"github.com/ipfs/kubo/test/cli/harness"
 	"github.com/stretchr/testify/assert"
 )
 
+const (
+	fixtureFile = "./fixtures/TestDagStat.car"
+	node1Cid    = "bafyreibmdfd7c5db4kls4ty57zljfhqv36gi43l6txl44pi423wwmeskwy"
+	node2Cid    = "bafyreie3njilzdi4ixumru4nzgecsnjtu7fzfcwhg7e6s4s5i7cnbslvn4"
+	fixtureCid  = "bafyreifrm6uf5o4dsaacuszf35zhibyojlqclabzrms7iak67pf62jygaq"
+)
+
+// The Fixture file represents a dag where 2 nodes of size = 46B each, have a common child of 53B
+// when traversing the DAG from the root's children (node1 and node2) we count (46 + 53)x2 bytes (counting redundant bytes) = 198
+// since both nodes share a common child of 53 bytes we actually had to read (46)x2 + 53 =  145 bytes
+// we should get a dedup ratio of 198/145 that results in approximatelly 1.3655173
+
 func TestDag(t *testing.T) {
 	t.Parallel()
-	t.Run("ipfs add, adds file", func(t *testing.T) {
+	t.Run("ipfs dag stat", func(t *testing.T) {
 		t.Parallel()
 		node := harness.NewT(t).NewNode().Init().StartDaemon()
-
-		output := node.IPFSAddStr("hello world")
-		output2 := node.IPFSAddStr("hello world 2")
-		output3 := node.IPFSAddStr("hello world 3")
-		assert.NotEqual(t, "", output)
-
-		stat := node.RunIPFS("dag", "stat", "-p", output, output2, output3)
-		str := stat.Stdout.String()
-		err := stat.Stderr.String()
-		assert.NotEqual(t, "", str)
+		// Import fixture
+		r, err := os.Open(fixtureFile)
 		assert.Nil(t, err)
-
+		defer r.Close()
+		err = node.IPFSDagImport(r, fixtureCid)
+		assert.Nil(t, err)
+		stat := node.RunIPFS("dag", "stat", "--progress=false", node1Cid, node2Cid)
+		str := stat.Stdout.String()
+		expected := "\nCID                                           \tBlocks         \tSize\nbafyreibmdfd7c5db4kls4ty57zljfhqv36gi43l6txl44pi423wwmeskwy\t2              \t53\nbafyreie3njilzdi4ixumru4nzgecsnjtu7fzfcwhg7e6s4s5i7cnbslvn4\t2              \t53\n\nSummary\nTotal Size: 145\nUnique Blocks: 3\nShared Size: 53\nRatio: 1.365517\n\n\n"
```

Comment:
Please use the \` syntax, this is very hard to me.

Reply:
sorry for that, I'm gonna make this more readable

Reply:
@Jorropo I'm having a hard time comparing the text output with a string literal in this test, I added a test using the json encoded output so that I can validate the deduplication behaviour. Is there any other way that I could test the text output?

Reply:
@arthurgavazza json is good.

For the text one: the point of golden tests is that the code of the test is very dumb.
You create a dataset ahead of time that show all the behaviour you want (here it would be a `.car` fixture) and then you `==` or `diff` the output with some hardcoded values you create at the same time you make your fixture.